syntax = "proto3";

service Lab6Service {
  rpc Add (AddMsg) returns (AddReply) {}
  rpc RawImage (RawImageMsg) returns (ImageReply) {}
  rpc DotProduct (DotProductMsg) returns (DotProductReply) {}
  rpc JsonImage (JsonImageMsg) returns (ImageReply) {}
}

message AddMsg {
  int32 a = 1;
  int32 b = 2;
}

message AddReply {
  int32 sum = 1;
}

message RawImageMsg {
  bytes img = 1;
}

message JsonImageMsg {
  string img = 1; // base64 string
}

message ImageReply {
  int32 width = 1;
  int32 height = 2;
}

message DotProductMsg {
  repeated float a = 1;
  repeated float b = 2;
}

message DotProductReply {
  float dotproduct = 1;
}